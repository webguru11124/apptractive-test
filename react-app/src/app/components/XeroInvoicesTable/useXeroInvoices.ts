import * as React from 'react';
import { gql, useSuspenseQuery } from '@apollo/client';
import {
  xeroGetInvoices as XEROGETINVOICES,
  XeroInvoice,
  XeroInvoiceStatus,
  xeroGetInvoiceCount as XEROGETINVOICECOUNT,
} from '../../graphql';
interface XeroGetInvoicesQueryResult {
  xeroGetInvoices: XeroInvoice[];
}
interface XeroGetInvoiceCountQueryResult {
  xeroGetInvoiceCount: number;
}

const GET_INVOICES_QUERY = gql`
  ${XEROGETINVOICES}
`;
const GET_INVOICE_COUNT_QUERY = gql`
  ${XEROGETINVOICECOUNT}
`;
const statues = Object.keys(XeroInvoiceStatus);

export const useXeroInvoices = () => {
  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = React.useState(5);
  const [statusSelected, setStatusSelected] = React.useState<string[]>(statues);

  const { data: xeroInvoices, refetch } =
    useSuspenseQuery<XeroGetInvoicesQueryResult>(GET_INVOICES_QUERY, {
      variables: {
        input: {
          page: page + 1,
          limit: rowsPerPage,
          statuses: statusSelected,
        },
      },
    });
  const handlePageChange = React.useCallback(
    (newPage: number) => {
      setPage(newPage);
      refetch();
    },
    [refetch]
  );

  const handleStatusSelectionChange = React.useCallback(
    (statuses: string[]) => {
      setStatusSelected(statuses);
    },
    []
  );

  const handleRowsPerPageChange = React.useCallback(
    (newRowsPerPage: number) => {
      setRowsPerPage(newRowsPerPage);
      setPage(0);
    },
    []
  );

  const { data: xeroInvoiceCount } =
    useSuspenseQuery<XeroGetInvoiceCountQueryResult>(GET_INVOICE_COUNT_QUERY, {
      variables: {
        input: {
          statuses: statusSelected,
        },
      },
    });
  return {
    data: xeroInvoices.xeroGetInvoices,
    page,
    setPage: handlePageChange,
    rowsPerPage,
    setRowsPerPage: handleRowsPerPageChange,
    totalCount: xeroInvoiceCount.xeroGetInvoiceCount,
    statusSelected,
    refetch,
    setStatusSelected: handleStatusSelectionChange,
  };
};
