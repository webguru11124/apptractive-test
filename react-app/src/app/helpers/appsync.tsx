import { AuthOptions } from 'aws-appsync-auth-link';
import cdkOutput from '../output.json';
import { Auth } from 'aws-amplify';
import {
  ApolloClient,
  ApolloLink,
  gql,
  HttpLink,
  InMemoryCache
} from '@apollo/client';
import { createAuthLink } from 'aws-appsync-auth-link';
import { createSubscriptionHandshakeLink } from 'aws-appsync-subscription-link';
import {
  isLoggedInVar,
  subInVar,
  typeDefs
} from '../graphql';

/**
 * Configure AppSync client
 */
const httpLink = new HttpLink({
  uri: cdkOutput.FULAppSyncAPIStack.GraphQLAPIURL
});

export const configureAppSyncClient = () => {
  const url = cdkOutput.FULAppSyncAPIStack.GraphQLAPIURL; // TODO: duplicated url?
  const region = 'us-east-1';

  const auth: AuthOptions = {
    type: 'AMAZON_COGNITO_USER_POOLS',
    jwtToken: async () =>
      (await Auth.currentSession()).getIdToken().getJwtToken()
    //credentials: async () => credentials, // Required when using IAM-based auth.
  };

  const link = ApolloLink.from([
    createAuthLink({ url, region, auth }),
    createSubscriptionHandshakeLink({ url, region, auth }, httpLink)
  ]);

  return new ApolloClient({
    link,
    cache: new InMemoryCache({
      typePolicies: {
        Query: {
          fields: {
            isLoggedIn: {
              read() {
                return isLoggedInVar();
              }
            },
            sub: {
              read() {
                return subInVar();
              }
            }
          }
        }
      }
    }),
    connectToDevTools: true,
    typeDefs: gql(typeDefs)
  });
};
