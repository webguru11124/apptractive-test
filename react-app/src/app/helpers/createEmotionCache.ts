import createCache from '@emotion/cache';
export type { EmotionCache } from '@emotion/cache';

export function createEmotionCache() {
  return createCache({ key: 'css' });
}
