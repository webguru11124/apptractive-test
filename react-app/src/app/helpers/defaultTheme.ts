import { PaletteMode, ThemeOptions } from '@mui/material';

export const defaultTheme = (mode: PaletteMode): ThemeOptions => ({
  components: {
    MuiCssBaseline: {
      styleOverrides: {
        html: {
          height: '100vh',
        },
        body: {
          height: '100vh',
        },
        '#root': {
          height: '100vh',
        },
      },
    },
  },
});
