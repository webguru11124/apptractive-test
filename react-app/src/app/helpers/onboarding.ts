import { OnboardingStatus, User } from '../graphql';

/**
 * Gets the next step of onboarding based on user type
 * @param user
 */
export const getNextOnboardingStep = (user: User) => {
  const { onboardingStatus } = user;
  switch (onboardingStatus) {
    case OnboardingStatus.NAME:
      return OnboardingStatus.PROFILE_IMAGE;
    case OnboardingStatus.PROFILE_IMAGE:
      return OnboardingStatus.INTERESTS;
    case OnboardingStatus.INTERESTS:
      return OnboardingStatus.PLANS;
    case OnboardingStatus.PLANS:
      return OnboardingStatus.COMPLETED;
    default:
      return OnboardingStatus.COMPLETED;
  }
};
