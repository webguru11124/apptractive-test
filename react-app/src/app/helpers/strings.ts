export function convertColumnNamesToDisplayText(inputString: string) {
  const newStr = inputString
    .replace(/([A-Z])/g, ' $1') // insert a space before all capital letters
    .replace(/^./, (str: string) => str.toUpperCase()) // capitalize the first letter of each word
    .replace(/([a-z0-9])([A-Z])/g, '$1 $2'); // insert a space between last lowercase & uppercase letter
  return newStr;
}
