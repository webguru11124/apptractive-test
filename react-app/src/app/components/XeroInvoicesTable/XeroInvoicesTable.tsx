import * as React from 'react';
import { XeroInvoice, XeroInvoiceStatus } from '../../graphql';
import { EnhancedTable } from '../EnhancedTable/EnhancedTable';
import { useXeroInvoices } from './hook/useXeroInvoices';
import MenuItem from '@mui/material/MenuItem';
import { Checkbox, ListItemText, Menu } from '@mui/material';

const columns: (keyof XeroInvoice)[] = [
  'status',
  'totalTax',
  'total',
  'invoiceNumber',
  'amountPaid',
];

const statues = Object.keys(XeroInvoiceStatus);

export function XeroInvoicesTable() {
  const {
    page,
    setPage,
    rowsPerPage,
    setRowsPerPage,
    data,
    order,
    orderBy,
    setOrder,
    setStatusSelected,
    statusSelected,
    refetch,
  } = useXeroInvoices();

  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
    refetch();
  };

  const handleStatusSelected = (status: string) => {
    const currentIndex = statusSelected.indexOf(status);
    const newStatusSelected = [...statusSelected];
    if (currentIndex === -1) {
      newStatusSelected.push(status);
    } else {
      newStatusSelected.splice(currentIndex, 1);
    }
    setStatusSelected(newStatusSelected);
  };

  return (
    <>
      <EnhancedTable<XeroInvoice>
        columns={columns}
        page={page}
        setPage={setPage}
        rowsPerPage={rowsPerPage}
        setRowsPerPage={setRowsPerPage}
        rows={data}
        order={order}
        orderBy={orderBy}
        onChangeOrder={setOrder}
        onFilterClick={handleClick}
      />
      <Menu
        id="basic-menu"
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        MenuListProps={{
          'aria-labelledby': 'basic-button',
        }}
      >
        {statues.map((status) => (
          <MenuItem
            key={status}
            value={status}
            onClick={() => handleStatusSelected(status)}
          >
            <Checkbox checked={statusSelected.indexOf(status) > -1} />
            <ListItemText primary={status} />
          </MenuItem>
        ))}
      </Menu>
    </>
  );
}
