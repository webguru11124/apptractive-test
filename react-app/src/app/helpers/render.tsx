import { MockedProvider } from '@apollo/client/testing';
import { CacheProvider } from '@emotion/react';
import { CssBaseline } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { ThemeProvider as EmotionThemeProvider } from '@emotion/react';
import { render } from '@testing-library/react';
import { PropsWithChildren, ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { DesignSystemContextProvider } from '../components';
import { getTheme } from '../constants/theme';
import { createEmotionCache } from './createEmotionCache';
import { defaultTheme } from './defaultTheme';

const defaultCache = createEmotionCache();
const theme = getTheme(defaultTheme('light'));

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return (
    <MockedProvider>
      <BrowserRouter>
        <CacheProvider value={defaultCache}>
          <EmotionThemeProvider theme={theme}>
            <ThemeProvider theme={theme}>
              <CssBaseline />
              <DesignSystemContextProvider theme={theme}>
                  {children}
              </DesignSystemContextProvider>
            </ThemeProvider>
          </EmotionThemeProvider>
        </CacheProvider>
      </BrowserRouter>
    </MockedProvider>
  );
};

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
