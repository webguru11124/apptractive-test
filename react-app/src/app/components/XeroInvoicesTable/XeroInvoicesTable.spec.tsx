import { render } from '../../helpers/render';

import { XeroInvoicesTable } from './XeroInvoicesTable';

describe('XeroInvoicesTable', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<XeroInvoicesTable />);
    expect(baseElement).toBeTruthy();
  });
});
