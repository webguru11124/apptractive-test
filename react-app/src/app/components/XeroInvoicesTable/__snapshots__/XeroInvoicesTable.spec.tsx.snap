// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`XeroInvoicesTable > renders XeroInvoicesTable component snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1mxz8qt-MuiPaper-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-pjjt8i-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 css-2pbtu6-MuiTypography-root"
          data-testid="invoices"
          id="tableTitle"
        >
          invoices
        </div>
        <button
          aria-label="filterList"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FilterListIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          aria-labelledby="tableTitle"
          class="MuiTable-root css-1gplja-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            data-testid="table-head"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-hbtnrh-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-label="select all desserts"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </th>
              <th
                aria-sort="ascending"
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  invoiceNumber
                  <span
                    class="MuiBox-root css-trfuzi"
                  >
                    sorted ascending
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  totalTax
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  total
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  amountPaid
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  status
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
        </table>
      </div>
      <div
        class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":r2:"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-controls=":r3:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r2: :r1:"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r1:"
              role="combobox"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0–0 of 0
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        densePadding
      </span>
    </label>
  </div>
</div>
`;
